ng/src/web/components/link/externallink.js

```diff
@@ -1,57 +1,123 @@
+*
+* Copyright:
+* Copyright (C) 2017 Greenbone Networks GmbH
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License
+* as published by the Free Software Foundation; either version 2
+* of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software
+* Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
 
 import React from 'react';
 
+import _ from 'gmp/locale.js';
+
 import {classes} from 'gmp/utils.js';
 
+import LinkConfirmationDialog from '../dialog/linkconfirmationdialog.js';
+
 import PropTypes from '../../utils/proptypes.js';
 
 import {withTextOnly} from './link.js';
 
-const ExternalLink = ({
-  to,
-  className,
-  children,
-  ...props
-}) => {
-
-  className = classes(className, 'external-link');
-
-  return (
-    <a
-      {...props}
-      href={to}
-      className={className}
-      rel="noopener noreferrer"
-      target="_blank">
-      {children}
-    </a>
-  );
-};
+class ExternalLink extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      dialogvisible: false,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleCloseDialog = this.handleCloseDialog.bind(this);
+  }
+
+  handleClick(event) {
+    event.preventDefault();
+    this.openDialog();
+  }
+
+  handleCloseDialog() {
+    this.setState({
+      dialogvisible: false,
+    });
+  }
+
+  handleOpenLink(url) {
+    window.open(url, '_blank', 'resizable=yes', 'noopener');
+  }
+
+  openDialog() {
+    this.setState({
+      dialogvisible: true,
+    });
+  }
+
+  render() {
+    const {
+      dialogvisible,
+    } = this.state;
+
+    let {
+      children,
+      className,
+      to,
+      href,
+      ...props
+    } = this.props;
+
+    if (href) {
+      to = href;
+    }
+
+    className = classes(className, 'external-link');
+
+    const dialogtitle = _('You are leaving GSA');
+    const dialogtext = _('This dialog will open a new window for {{- to}} ' +
+      'if you click on "follow link". Following this link is on your own ' +
+      'responsibility. Greenbone does not endorse the content you will ' +
+      'see there.', {to});
+    return (
+      <span>
+        <a
+          {...props}
+          href={to}
+          className={className}
+          onClick={this.handleClick}
+        >
+          {children}
+        </a>
+        <LinkConfirmationDialog
+          visible={dialogvisible}
+          onClose={this.handleCloseDialog}
+          onResumeClick={() => this.handleOpenLink(to)}
```

Comment:
Not necessary to create an arrow function to pass the to prop to the handler. The handler can use this.props.to instead.